src/csharp/Extensions/Microsoft.Spark.Extensions.DotNet.Interactive/PackageHelper.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.Interactive;
+using Microsoft.DotNet.Interactive.Utility;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Utils;
+
+namespace Microsoft.Spark.Extensions.DotNet.Interactive
+{
+    internal static class PackagesHelper
+    {
+        private static readonly HashSet<string> s_filesCopied = new HashSet<string>();
+        private static ulong s_metadataCounter = 0;
+
+        internal static void GenerateAndAddFiles(
+            string writePath,
+            Action<string> fileAction)
+        {
+            IEnumerable<NuGetPackage> nugetPackagesToCopy = GetPackagesToCopy();
+
+            var assemblyProbingPaths = new List<string>();
+            var nativeProbingPaths = new List<string>();
+            var nugetMetadata = new List<DependencyProviderUtils.NuGetMetadata>();
+
+            foreach (NuGetPackage package in nugetPackagesToCopy)
+            {
+                ResolvedPackageReference resolvedPackage = package.ResolvedPackage;
+                foreach (FileInfo asmPath in resolvedPackage.AssemblyPaths)
+                {
+                    assemblyProbingPaths.Add(
+                        GetPathRelativeToPackages(
+                            asmPath.FullName,
+                            resolvedPackage.PackageRoot));
+                }
+
+                foreach (DirectoryInfo probePath in resolvedPackage.ProbingPaths)
+                {
+                    nativeProbingPaths.Add(
+                        GetPathRelativeToPackages(
+                            probePath.FullName,
+                            resolvedPackage.PackageRoot));
+                }
+
+                FileInfo nugetFile = package.NuGetFile;
+                if (!IsPathValid(nugetFile.FullName))
+                {
+                    // Copy the nuget file to a location that works with 
+                    // SparkContext.AddFiles(..)
```

Comment:
why is this? (e.g., why wouldn't it work)?

Reply:
The original nupkg file can be in a location like,   `C:\Users\user with spaces\.nuget\packages\package.name\version\package.name.version.nupkg` and that would not work with SparkContext.AddFiles(..) with a spark version lower than 3.